components/modules/net.c

```diff
@@ -982,9 +995,10 @@ static void ldnsfound_cb (lua_State *L, lnet_userdata *ud, ip_addr_t *addr) {
     lua_call(L, 2, 0);
   }
   ud->client.wait_dns --;
-  if (ud->pcb && ud->type == TYPE_TCP_CLIENT && ud->tcp_pcb->state == CLOSED) {
-    tcp_connect(ud->tcp_pcb, addr, ud->tcp_pcb->remote_port, net_connected_cb);
-  } else if (!ud->pcb && ud->client.wait_dns == 0) {
+  if (ud->netconn && ud->type == TYPE_TCP_CLIENT && !ud->client.connecting) {
+    ud->client.connecting = true;
+    netconn_connect(ud->netconn, addr, ud->port);
+  } else if (!ud->netconn && ud->client.wait_dns == 0) {
```

Comment:
Ah, I misread that line! I somehow saw it as `!ud->netconn && ud->netconn.wait_dns` and thus took the first part to be a null pointer guard.